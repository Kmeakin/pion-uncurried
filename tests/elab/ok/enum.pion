enum YesOrNo {
    Yes(_: Bool),
    No(_: Type),
}

enum Maybe(T: Type) {
    None,
    Some(x: T),
}

enum Either(A: Type, B: Type) {
    Left(x: A),
    Right(y: B),
}

enum Result(T: Type, E: Type) {
    Ok(t: T): Result(T, E),
    Err(e: E): Result(T, E),
}

let yorn = YesOrNo;
let y = Yes;
let n = No;

let m = Maybe;
let n = None;
let s = Some;

let e = Either;
let l = Left;