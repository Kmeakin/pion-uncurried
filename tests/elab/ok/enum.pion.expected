enum YesOrNo: Type {
    Yes(_: Bool): YesOrNo,
    No(_: Type): YesOrNo,
}

enum Maybe(T: Type): Type {
    None: Maybe(T),
    Some(x: T): Maybe(T),
}

enum Either(A: Type, B: Type): Type {
    Left(x: A): Either(A, B),
    Right(y: B): Either(A, B),
}

enum Result(T: Type, E: Type): Type {
    Ok(t: T): Result(T, E),
    Err(e: E): Result(T, E),
}

let yorn: fn() -> Type = YesOrNo;

let y: fn(_: Bool) -> YesOrNo = Yes;

let n: fn(_: Type) -> YesOrNo = No;

let m: fn(T: Type) -> Type = Maybe;

let n: fn(T: Type) -> Maybe(T) = None;

let s: fn(T: Type, x: T) -> Maybe(T) = Some;

let e: fn(A: Type, B: Type) -> Type = Either;

let l: fn(A: Type, B: Type, x: A) -> Either(A, B) = Left;
